Get SPEC file to build with 28.2 and FC35

derekatkins opened the pull request:
This PR gets the SPEC file to work properly with the release tarball.  Specifically, I needed to:

* Change the Source tarball from GZ to XZ
* always depend on protobuf-c (configure exits if it's not found, even without chromecast)
* the build system installed owntone.service into /etc, which is wrong.  Exclude that location and use the correct location.

With these changes, the RPM builds on FC35.

ejurgensen commented:
Thanks for the PR! Looks good to me. I think there might be some more updates that I'm hoping you could include? I see it has been a while since the last update of the spec file, and there have been some changes to owntone configuration that haven't been reflected in the current spec file:

1. --with-libplist, --with-libsodium and --with-libcurl no longer exist as options (the libraries are now hard requirements), so can be deleted
2. "make install" will now install a systemd service file, so probably no need to do it manually as in the %install section. If the default location that it gets installed to isn't good then there is a --with-systemddir option.
3. Since the spec file seems to take care of the config file, then perhaps it is relevant to configure with --disable-install_conf_file, otherwise "make install" will also try to install a config file
4. --with-daapd-user and --with-daapd-group are now --with-user and --with-group

I'm not into spec files/building rpms, so it would be great if you would be up to also including these updates and testing that it still works.

derekatkins commented:
Hi,
> Thanks for the PR! Looks good to me. I think there might be some more updates that I'm hoping you could include? I see it has been a while since the last update of the spec file, and there have been some changes to owntone configuration that haven't been reflected in the current spec file:
> 
>     1. --with-libplist, --with-libsodium and --with-libcurl no longer exist as options (the libraries are now hard requirements), so can be deleted

Okay.  Those are easily removed.

>     2. "make install" will now install a systemd service file, so probably no need to do it manually as in the %install section. If the default location that it gets installed to isn't good then there is a --with-systemddir option.

The issue here is that "make install" puts it into /etc/systemd/system instead of /usr/lib/systemd/system, which would not be acceptable in general.  To work around that I install the (same) file into the correct location and ignore the one installed by "make install".

>     3. Since the spec file seems to take care of the config file, then perhaps it is relevant to configure with --disable-install_conf_file, otherwise "make install" will also try to install a config file

I don't see where the spec file installs the .conf file?  It just distributes the one installed by "make install"

>     4. --with-daapd-user and --with-daapd-group are now --with-user and --with-group

Changed.  I will push those changes up to this branch (and update this PR) momentarily.
 
> I'm not into spec files/building rpms, so it would be great if you would be up to also including these updates and testing that it still works.

Do these changes apply to 28.2 or were those changes made after the release?

derekatkins commented:
@ejurgensen -- One thing I did notice, by the way, is that even though the server was not compiled with spotify or lastfm support (I did not use --with-spotify or --with-lastfm), the build still included them and they show up as options in the server.  Are those not optional?

ejurgensen commented:
> The issue here is that "make install" puts it into /etc/systemd/system instead of /usr/lib/systemd/system, which would not be acceptable in general.

Might be cleaner to use `--with-systemddir` to specify the dir? Then line 83 + 84 could be removed. Or if not, use `--disable_install_systemd` to avoid the unused duplicate file in `/etc`?

> I don't see where the spec file installs the .conf file?

Yes, maybe I read the file wrong. It looks like line 107 `%config(noreplace) %{_sysconfdir}/owntone.conf` might install the file, especially since it has the noreplace flag. If `--disable-install_conf_file` isn't given to configure, then `make install` will install the conf file from the build dir to `/etc` (or `/usr/local/etc` depending on sysconfdir), which might not be necessary or desired if line 107 is doing it the proper Fedora way.

> Do these changes apply to 28.2 or were those changes made after the release?

Yes, I think they all apply to 28.2.

> Are those not optional?

They are still optional, but before before they were disabled by default, now they are enabled. That is because before their dependencies are no longer special. If you want, you can still disable with `--disable-spotify` and `--disable-lastfm`.

ejurgensen commented:
> I tried --with-systemdir and it fix the issue. I can try the --disable_install_systemd to see if that makes a difference.

Sounds good. If `--with-systemddir` works then you don't need to try `--disable_install_systemd`, they shouldn't be used together.

> IMHO it would be better to just remove those flags completely and simplify the spec file. Do you concur?

Yes, I agree.

derekatkins commented:
Hi,

> > I tried --with-systemdir and it fix the issue. I can try the --disable_install_systemd to see if that makes a difference.
> 
> Sounds good. If `--with-systemddir` works then you don't need to try `--disable_install_systemd`, they shouldn't be used together.

Sorry, this was a typo..  I fixed it above.  I meant to say it /did not/ fix the problem.  I'll test if disable works as expected.
 
> > IMHO it would be better to just remove those flags completely and simplify the spec file. Do you concur?
> 
> Yes, I agree.

I've removed those flags.

derekatkins commented:
Using --disable-install_systemd worked, so I've updated the spec accordingly.

derekatkins commented:
Oh, I think I was using the wrong configure option for the systemd dir.  You're right, I should use -with-systemddir instead of what I'm doing.  I've verified that that works as expected.

At this point I think it is ready for you; I think I've hit all your questions/comments.

ejurgensen commented:
Looks good, merged. Thanks a lot!
